### PR TITLE
Endpoint for my tasks

### LINKED ISSUE #1597: Create user tasks API view
Implement API view that lists the tasks of all projects mapped or validated by a user. Allow to apply filters by date, project id, mapping/validating, plus ordering options.

### PR DESCRIPTION
**Pull request by fitoria**
fixes #1597 was PR #1684

How to test:

Authenticated ajax request to /api/v1/user/tasks

Filters as get params, available filters: status, project_id, min_action_date, max_action_date

Sample request url: /api/v1/user/tasks?status=READY&project_id=2&min_action_date=2019-04-9&max_action_date=2019-04-10

Sample response:
```
{
    "tasks": [
        {
            "taskId": 9,
            "projectId": 1,
            "taskStatus": "READY",
            "taskHistory": [],
            "taskAnnotation": null,
            "isUndoable": false,
            "autoUnlockSeconds": 7200
        },
        {
            "taskId": 16,
            "projectId": 1,
            "taskStatus": "READY",
            "taskHistory": [],
            "taskAnnotation": null,
            "isUndoable": false,
            "autoUnlockSeconds": 7200
        }
    ]
}
```

### CODE REVIEW

**Comment by willemarcel** [comment]
Thank you very much, @fitoria! Sorry for the delay on reviewing it.

- Instead of min_action_date and max_action_date, we could use `start_date` and `end_date`.
- It would be good if we could sort by action_date
- We could also include the `action_date` field on the response

File: server/__init__.py [hunk 1]
```diff
@@ -476,6 +477,7 @@ def add_api_endpoints(app):
     api.add_resource(
         UsersQueriesUsernameAPI, "/api/v2/users/queries/<string:username>/"
     )
+    api.add_resource(UserTasksAPI, "/api/v2/user/tasks/")
```
Comment (xamanu) [inline server/__init__.py]: Should probably be `/api/v2/users/me/tasks/`.

File: server/api/users/resources.py [hunk 2]
```diff
@@ -193,3 +194,90 @@ def get(self, username):
             error_msg = f"User GET - unhandled error: {str(e)}"
             current_app.logger.critical(error_msg)
             return {"Error": "Unable to fetch matching users"}, 500
+
+
+class UserTasksAPI(Resource):
```
Comment (xamanu) [inline server/api/users/resources.py]: Should be `UsersTasksAPI` and be in a `tasks.py`

File: server/__init__.py [hunk 3]
```diff
@@ -215,6 +215,7 @@ def add_api_endpoints(app):
         UsersAllAPI,
         UsersQueriesUsernameAPI,
         UsersQueriesUsernameFilterAPI,
+        UserTasksAPI,
```
Comment (xamanu) [inline server/__init__.py]: Should be `UsersTasksAPI` and live in `server/api/users/tasks.py`

**Comment by xamanu** [comment]
This is the base for #1599

**Comment by xamanu** [comment]
As per request of @willemarcel, please extend to filter on `taskStatus` (e.g. = `locked`).

**Comment by willemarcel** [comment]
Thanks, @fitoria ! 

"taskHistory", "isUndoable", "autoUnlockSeconds" are not necessary, but it's not a big problem to have them. Although `taskHistory` always returned an empty array for me.

Do you think it would be complex if we add the user of the last task action? If it is complex or it requires too much from the database, I prefer to not having them.

**Comment by xamanu** [comment]
There are also some conflicts, please rebase on latest `develop`. Thanks.

**Comment by willemarcel** [comment]
I just saw that this PR is also related to https://github.com/hotosm/tasking-manager/issues/1597, so maybe we could change the url from `/users/me/tasks` to `users/:username/tasks`, so it will handle the two needs that we have

**Comment by xamanu** [comment]
Shouldn't we use the `user_id` in the good old Restful way?

**Comment by willemarcel** [comment]
Thank you, @fitoria! It seems good to me.

**Comment by willemarcel** [comment]
I changed the URL param to user_id instead of username, @xamanu

**Pull request merged**